init-tests-after-clone.sh

```diff
@@ -1,18 +1,73 @@
-#!/usr/bin/env bash
+#!/bin/sh
 
-set -e
+set -eu
 
-if [[ -z "$TRAVIS" ]]; then
-  read -rp "This operation will destroy locally modified files. Continue ? [N/y]: " answer
-  if [[ ! $answer =~ [yY] ]]; then
-    exit 2
-  fi
+fallback_repo_for_tags='https://github.com/gitpython-developers/GitPython.git'
+
+ci() {
+    # For now, check just these, as a false positive could lead to data loss.
+    test -n "${TRAVIS-}" || test -n "${GITHUB_ACTIONS-}"
+}
+
+no_version_tags() {
+    test -z "$(git tag -l '[0-9]*' 'v[0-9]*')"
+}
+
+warn() {
+    if test -n "${GITHUB_ACTIONS-}"; then
+        printf '::warning ::%s\n' "$*" >&2  # Annotate workflow.
+    else
+        printf '%s\n' "$@" >&2
+    fi
+}
+
+if ! ci; then
+    printf 'This operation will destroy locally modified files. Continue ? [N/y]: ' >&2
+    read -r answer
+    case "$answer" in
+    [yY])
```

Comment:
This is probably for portability or because `sh` can't portably match against globs or regex.

Reply:
The `[[` keyword is a `bash`-ism. `[` and `test` are POSIX, but they don't support the `=~` operator for matching regular expressions. So that's why I replaced an `if` check on a `[[` command using the `=~` operator, with `case`.

`case` matches globs, and conveniently `[yY]` has the same meaning as a regular expression and as a glob.